### DISCUSSION TITLE
`IAsyncEnumerable` - Any support for this? I don't see any mention of IAsyncEnumerable in this repository at all

### DISCUSSION THREAD

**Discussion by jeffward01** [vkhorikov/CSharpFunctionalExtensions#470] [Q&A] [answered]
Hey all,

I wanted to check and see if IAs is supported. I see in `language-ext` they are doing this:

> The netstandard2.1 release supports `IAsyncEnumerable<A>` for `OptionAsync<A>` and `EitherAsync<A>`. This is the first baby-step towards leveraging some of the newer features of C# and .NET Core. - [Source](**url**)

That would equate to something like `ResultAsync<T,E>` in `CSharpFunctionalExtensions` - if it was a one-for-one comparison.

# Question

How should I return a `IAsyncEnumerable` from a method?

Most of my methods return `Result<Maybe<T>,E>>` or `Result<T,E>` - using this pattern, what do you suggest?

**Comment by vkhorikov** [comment, marked as answer]
No, there's no support for `IAsyncEnumerable` or similar. As far as I understand, that's for IO/out-of-process dependencies, which this library doesn't work with.

I don't see the need for `ResultAsync` either because all extension methods (sync or async) work with the `Result` class itself.

Since `IAsyncEnumerable` should be the upper-most monad, you'd have to do something line `IAsyncEnumerable<Maybe<T>>`. This doesn't really work well, so I'd recommend to first process the `IAsyncEnumerable` and then convert the whatever result to `Maybe` or `Result`.

**Reply by jeffward01** [reply]
This makes alot of sense, thanks! I was having some complications processing `IEnumerable` regarding using a `yield` within an iterator. Because my return type was `Result<IEnumerable<TEntit>, Error>` I struggled to use `yield` at all - , so I did not touch the `IAsynEnumerable.

Thanks for this
